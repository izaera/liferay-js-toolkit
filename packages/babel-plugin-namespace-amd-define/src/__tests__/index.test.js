import * as babel from 'babel-core';
import plugin from '../index';

it('namespaces unqualified define calls', () => {
	const source = `
	define([], function(){})
	`;

	const {code} = babel.transform(source, {
		plugins: [plugin],
	});

	expect(code).toMatchSnapshot();
});

it('does not namespace already qualified define calls', () => {
	const source = `
	Other.Namespace.define([], function(){})
	`;

	const {code} = babel.transform(source, {
		plugins: [plugin],
	});

	expect(code).toMatchSnapshot();
});

it('namespaces references to global define identifier', () => {
	const source = `
	if (typeof define === "function" && define.amd) {
		console.log('UMD!');
	}
	`;

	const {code} = babel.transform(source, {
		plugins: [plugin],
	});

	expect(code).toMatchSnapshot();
});

describe('does not namespace references to local define identifier', () => {
	it('when it is a variable', () => {
		const source = `
		let define = 'x';

		if(true) {
			if(true) {
				console.log(define);
			}
	    }
		`;

		const {code} = babel.transform(source, {
			plugins: [plugin],
		});

		expect(code).toMatchSnapshot();
	});

	it('when it is a function parameter', () => {
		const source = `
		function fn(define) {
			console.log(define);
		};
		`;

		const {code} = babel.transform(source, {
			plugins: [plugin],
		});

		expect(code).toMatchSnapshot();
	});

	it('when it is a function', () => {
		const source = `
		function define(x) {
			console.log(define);
		}
		`;

		const {code} = babel.transform(source, {
			plugins: [plugin],
		});

		expect(code).toMatchSnapshot();
	});
});
