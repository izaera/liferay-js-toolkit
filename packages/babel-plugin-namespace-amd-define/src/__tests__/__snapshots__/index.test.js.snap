// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`does not namespace already qualified define calls 1`] = `
"
Other.Namespace.define([], function () {});"
`;

exports[`does not namespace references to local define identifier when it is a function 1`] = `
"
function define(x) {
	console.log(define);
}"
`;

exports[`does not namespace references to local define identifier when it is a function parameter 1`] = `
"
function fn(define) {
	console.log(define);
};"
`;

exports[`does not namespace references to local define identifier when it is a variable 1`] = `
"
let define = 'x';

if (true) {
	if (true) {
		console.log(define);
	}
}"
`;

exports[`namespaces references to global define identifier 1`] = `
"
if (typeof Liferay.Loader.define === \\"function\\" && Liferay.Loader.define.amd) {
	console.log('UMD!');
}"
`;

exports[`namespaces unqualified define calls 1`] = `
"
Liferay.Loader.define([], function () {});"
`;
